Runtime/Scripts/MobiledgeXWebSocketClient.cs

```diff
@@ -149,31 +172,48 @@ public async Task<string> Receive(UInt64 maxSize = MAXREADSIZE)
                         Console.Error.WriteLine("Warning: Message is bigger than expected!");
                     }
                 } while (!chunkResult.EndOfMessage);
+                
                 ms.Seek(0, SeekOrigin.Begin);
-
-                // Looking for UTF-8 JSON type messages.
-                if (chunkResult.MessageType == WebSocketMessageType.Text)
-                {
-                    return StreamToString(ms, Encoding.UTF8);
-                }
-
+                return new Dictionary<WebSocketMessageType, MemoryStream>(){ [chunkResult.MessageType]= ms };
             }
 
-            return "";
+            return null;
         }
 
+        /// <summary>
+        /// RunReceive is used in receive thread
+        /// RunReceive receives websocket messages asynchronously and add it to either receiveQueue or receiveQueueBinary ...
+        /// dependent on the MessageType
+        /// </summary>
         public async void RunReceive()
         {
             Debug.Log("WebSocket Message Receiver looping.");
-            string result;
+            Dictionary<WebSocketMessageType, MemoryStream> response = new Dictionary<WebSocketMessageType, MemoryStream>(1);
             while (run)
             {
                 //Debug.Log("Awaiting Receive...");
-                result = await Receive();
-                if (result != null && result.Length > 0)
+                response = await Receive();
+                if (response != null && response.Keys.Count > 0)
                 {
                     //Debug.Log("Received: " + result);
-                    receiveQueue.Enqueue(result);
+                    if (response.Keys.First() == WebSocketMessageType.Text)
```

Comment:
how does the developer set the receive type?

Reply:
The developer doesn't need to set the receive type, The receive type is dedicated by the server.

Reply:
as discussed
 chunkResult is of type WebSocketReceiveResult not byte[]
WebSocketReceiveResult chunkResult >> WebSocketReceiveResult has MessageType field